src/components/dropdowns/primary-action-dropdown/primary-action-dropdown.js

```diff
@@ -211,89 +208,82 @@ Option.defaultProps = {
   when the dropdown trigger itself is clicked. Otherwise it would open and close
   immediately.
  */
-export class PrimaryActionDropdown extends React.PureComponent {
-  static displayName = 'PrimaryActionDropdown';
-
-  static propTypes = {
-    children: PropTypes.node.isRequired,
-  };
-
-  state = {
-    isOpen: false,
-  };
+const PrimaryActionDropdown = props => {
+  const ref = React.useRef();
+  const [isOpen, setIsOpen] = React.useState(false);
 
-  ref = React.createRef();
-
-  handleOpen = () => {
-    this.setState({ isOpen: true });
-  };
+  const handleGlobalClick = React.useCallback(
+    event => {
+      const dropdownButton = ref.current;
+      if (
+        event.target !== dropdownButton &&
+        !dropdownButton.contains(event.target)
+      ) {
+        setIsOpen(false);
+      }
+    },
+    [ref]
```

Comment:
I think I saw an issue or something related to defining DOM elements (refs) as dependencies. I’ll see if I can find more about it but maybe we should double check if it makes sense to do so.